src/utils/getCalendarPlans.ts

```diff
@@ -0,0 +1,149 @@
+  if (viewEnd.isAfter(currentEnd)) {
+    viewEnd = currentEnd.clone();
+  }
+
+  // 항상 절대적인 날짜 차이를 반환해야함
+  const weekDayDiff =
+    viewEnd.clone().endOf('w').diff(viewStart.clone().startOf('w'), 'days') + 1;
+  const weekDiff = Math.ceil(weekDayDiff / 7);
+
+  // 주차를 돔
+  for (let i = 0; i < weekDiff; i++) {
+    let newViewStart = null;
+    let newViewEnd = null;
+
+    // 시작날짜의 주차와 종료날짜의 주차 비교
+    if (weekDiff === 1) {
+      newViewStart = viewStart;
+      newViewEnd = viewEnd;
+    } else {
+      if (i === 0) {
+        // 첫주
+        newViewStart = viewStart;
+        newViewEnd = viewStart.clone().endOf('w');
+      } else if (i === weekDiff - 1) {
+        // 마지막주
+        newViewStart = viewEnd.clone().startOf('w');
+        newViewEnd = viewEnd;
+      } else {
+        // 중간주
+        newViewStart = viewStart.clone().add(i, 'w').startOf('w');
+        newViewEnd = viewStart.clone().add(i, 'w').endOf('w');
+      }
+    }
+
+    // 현재 달의 첫번째 주를 기준으로 몇번째 주인지
+    const weekOfMonth =
+      (newViewStart.clone().endOf('w').diff(currentStart, 'days') + 1) / 7 - 1;
+
+    // 일정이 표시될 날짜의 차이
+    const dayDiff =
+      Math.abs(newViewEnd.clone().diff(newViewStart.clone(), 'days')) + 1;
+
+    const currentStartTime = moment(startTime).startOf('d');
+    const currentEndTime = moment(endTime).endOf('d');
+
+    const dayDiffToCurrent = Math.abs(
+      currentEndTime.diff(currentStartTime.clone(), 'days'),
+    );
+
+    result.push({
+      id,
+      dayDiff,
+      weekOfMonth,
+      dayDiffToCurrent,
+      viewStart: newViewStart.clone(),
+      viewEnd: newViewEnd.clone(),
+      startTime: currentStartTime,
+      endTime: currentEndTime,
+      dayOfWeek: newViewStart.day() + 1,
+      plan: plan.id === -1 ? null : plan,
+    });
+  }
+
+  return result;
+};
+
+const setIndexAndReturnPlanViews = (viewPlans: IViewPlanInfo[]) => {
+  const planViewsToCalendar: IIndexableViewPlan[][] = Array.from(
+    { length: 6 },
+    () => Array.from({ length: 7 }, () => ({})),
+  );
+
+  viewPlans.forEach((planView) => {
+    const { weekOfMonth, dayOfWeek, dayDiff } = planView;
+    let index = 0;
+
+    for (let l = 0; l < dayDiff; l++) {
+      const dayInfo = planViewsToCalendar[weekOfMonth][dayOfWeek + l - 1];
+
+      while (dayInfo[index]?.id !== planView.id) {
+        if (!dayInfo[index]) {
+          dayInfo[index] = planView;
+        } else {
+          index++;
+        }
+      }
+    }
+  });
+
+  return planViewsToCalendar;
+};
+
+const sortPlansCallback = (a: IViewPlanInfo, b: IViewPlanInfo) => {
+  return (
+    a.startTime.diff(b.startTime) ||
+    b.dayDiffToCurrent - a.dayDiffToCurrent ||
+    a.id - b.id
+  );
+};
+
+const getCalendarPlans = (
+  plans: IPlan[],
+  startDate: TDateYMD,
+  endDate: TDateYMD,
+) => {
+  const planViewsArr: IViewPlanInfo[] = plans
+    .map((el) => getViewPlans(el, startDate, endDate))
+    .flat()
+    .sort(sortPlansCallback);
+
+  const planViewsToCalendar: IIndexableViewPlan[][] =
```

Comment:
Calendar를 인덱싱해서 보여주기 때문에 `IIndexableViewPlan`을 사용하신 것 같습니다.

다만, 타입의 이름이 눈에 안띌 뿐더러 인덱스와 일정을 매핑만하는 객체에 대해 다시 한번 생각하게 되는 것 같습니다.
아래와 같이 `Map` 객체를 이용해 표현해서 가독성을 올려보는 것은 어떤가요?

```suggestion
  const planViewsToCalendar: Map<number, IViewPlanInfo>[][] =
```

Reply:
`src/types/index.ts` 파일에 코멘트했지만, `IIndexableViewPlan`은 결국 `IViewPlanInfo`를 배열처럼 사용하고 있기 때문에 굳이 타입을 더 하나 만든다면

```ts
type TViewPlanInfoArr = IViewPlanInfo[];

const planViewsToCalendar: TViewPlanInfoArr[][] = 
...
```

처럼 사용하거나, 굳이 타입을 새로 안만들어도 된다면

```ts
const palnViewsToCalendar: ViewPlanInfo[][][] = 
...
```
처럼 사용해도 될 것 같습니다. 이 부분은 세 명이 함께 의논해봐야 할 것 같네요.

Reply:
![image](https://user-images.githubusercontent.com/32933980/233285950-0e67636d-6f47-4533-bd0d-8a5ff10a45f9.png)

이처럼 중간에 존재하지 않으면 for 문을 돌 때 건너뛰기 때문에 배열로 사용해도 된다고 생각합니다!

Reply:
1. Map을 타입에 할당할 경우 기존 `{}` 값이 할당되지 않는 문제가 있습니다.
2. 따라서 배열을 할당하여 index로 접근하는 것으로 진행하겠습니다.